CauseContract.sol

```diff
@@ -50,29 +86,55 @@ contract CauseContract {
     }
 
     function retrieveInfo() public view returns (ContractInfo memory) {
-        return ContractInfo(id, admin, incoming, outgoing, contractAddress);
+        return ContractInfo(id, admin, incoming, outgoing, contractAddress, causeTotal, endCause, emailAddress, causeDescription, websiteURL, thumbnail);
     }
 
-    uint256 public transactionFee;
-    
-    function donate() public payable returns (uint256) {
+
+    function donate() public payable returns (bool) {
+
         require(msg.value > 0, "You must send some Ether");
+        require(endCause == 1, "This cause has ended, your funds have been returned");
+
+        uint256 gasStart = gasleft();
+
+        transactionFee = (msg.value*BASIS_POINTS) / 1000; // Transaction fee of 5bps (by default)
+        
+
+        blockChange.transfer(transactionFee);
```

Comment:
Which method of fund transfer did we decide was better, .transfer() or .call() - could this be updated

Reply:
I believe call is best in this instance, but for our purposes they are equivalent, call just gives us more flexibility.